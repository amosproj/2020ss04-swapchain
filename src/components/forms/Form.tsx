import React, { ReactElement } from "react"
interface Props {
  main: ReactElement
  footer: ReactElement
}

export const Form = (props: Props) => {
  return (
    <div className="flex flex-col items-center w-full bg-white border rounded shadow-lg">
      <div className="flex flex-col justify-center w-full p-4 space-y-4">{props.main}</div>
      <footer className="flex justify-center w-full p-4 bg-gray-100 border-t border-gray-200">{props.footer}</footer>
    </div>
  )
}
