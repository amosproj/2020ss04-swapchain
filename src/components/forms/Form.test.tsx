import React from "react"
import { Form } from "./Form"
import renderer from "react-test-renderer"

it("renders correctly", () => {
  const tree = renderer.create(<Form main={<h1>hello</h1>} footer={<h1>world</h1>}></Form>).toJSON()
  expect(tree).toMatchSnapshot()
})
