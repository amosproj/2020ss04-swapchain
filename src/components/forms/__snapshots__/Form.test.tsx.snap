// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-col items-center w-full bg-white border rounded shadow-lg"
>
  <div
    className="flex flex-col justify-center w-full p-4 space-y-4"
  >
    <h1>
      hello
    </h1>
  </div>
  <footer
    className="flex justify-center w-full p-4 bg-gray-100 border-t border-gray-200"
  >
    <h1>
      world
    </h1>
  </footer>
</div>
`;
